src/hera/workflows/script.py

```diff
@@ -555,7 +544,7 @@ def _extract_return_annotation_output(source: Callable) -> List:
     return_annotation = inspect.signature(source).return_annotation
     origin_type = get_origin(return_annotation)
     annotation_args = get_args(return_annotation)
-    if origin_type is Annotated:
+    if _type_util.is_annotated(return_annotation):
         output.append(annotation_args)
```

Comment:
I think this is also slightly buggy old code due to the assumption of only having 1 piece of metadata that must be an Artifact or Parameter. We can use the new `get_annotated_metadata(return_annotation, (Artifact, Parameter))` here.

```suggestion
    if param_or_artifact := get_annotated_metadata(return_annotation, (Artifact, Parameter)):
        output.append(param_or_artifact)
```

Reply:
Could we implement this change @jeongukjae ?